platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -86,19 +86,22 @@ SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
       try {
         uploadCoreConfiguration(collection, client);
       } catch (SolrFactoryException e) {
-        LOGGER.debug("Solr({}): Unable to upload configuration to Solr Cloud", collection, e);
+        LOGGER.debug("Solr({}): Unable to upload configuration to SolrCloud", collection, e);
         return null;
       }
 
       try {
         createCollection(collection, client);
       } catch (SolrFactoryException e) {
-        LOGGER.debug("Solr({}): Unable to create collection on Solr Cloud", collection, e);
+        LOGGER.debug("Solr({}): Unable to create collection on SolrCloud", collection, e);
         return null;
       }
 
       client.setDefaultCollection(collection);
       return closer.returning(client);
+    } catch (LinkageError | Exception e) {
```

Comment:
Is there something more specific we can catch besides `Exception`?

Reply:
I think we had something more specific before but any unhandled exceptions from the factory will cause failsafe to exit and stop retrying. So that is why this is as aggressive as it is on catching exceptions.